sql/ksp/ksp.sql

```diff
@@ -26,10 +26,53 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  ********************************************************************PGR-GNU*/
 
 --v2.6
+CREATE FUNCTION pgr_ksp(
+    TEXT, -- edges_sql (required)
+    BIGINT, -- from_vids (required)
+    BIGINT,   -- to_vids (required)
+    INTEGER, -- K (required)
+
+    directed BOOLEAN DEFAULT true,
+    heap_paths BOOLEAN DEFAULT false,
+
+    OUT seq INTEGER,
+    OUT path_id INTEGER,
+    OUT path_seq INTEGER,
+    OUT node BIGINT,
+    OUT edge BIGINT,
+    OUT cost FLOAT,
+    OUT agg_cost FLOAT)
```

Comment:
ok so additional columns have to be added in older SQL. because i cannot create new SQL with same signature(i tried it gives me error that pgr_ksp already exists something like that)

Reply:
Correct. And that's why the migration is needed for the older function - https://docs.pgrouting.org/latest/en/migration.html (you also need to add in the documentation on how to migrate the old function). We try not to make any breaking API change, but if made, the changes should be very minimal. You can look at `pgr_dijkstra (One to One)` here - earlier it had no `start_vid` or `end_vid` in the output column, then later we added it.

In your case, if the end-user used this query `SELECT colA, colB, ... from your_function` for his/her application, it will have no impact in the new release. But, in case the user used `SELECT * from your_function` in the old release, and the upgrade pgRouting, it can have impact, especially if that query was used inside a function. Hence, we write a migration guide informing the user of the same - `If needed filter out the added columns`.